src/org/opendatakit/briefcase/ui/automation/AutomationPanel.java

```diff
@@ -34,33 +38,76 @@
   private final FormCache formCache;
   private final BriefcasePreferences appPreferences;
 
-  public AutomationPanel(AutomationPanelForm view, TransferForms forms, BriefcasePreferences appPreferences, FormCache formCache) {
+  private Optional<Source<?>> pullSource;
+  private Optional<Source<?>> pushSource;
+
+  public AutomationPanel(AutomationPanelForm view, TransferForms forms, BriefcasePreferences tabPreferences, BriefcasePreferences appPreferences, FormCache formCache) {
     AnnotationProcessor.process(this);
     this.view = view;
     this.forms = forms;
     this.formCache = formCache;
     this.appPreferences = appPreferences;
 
+    // Read prefs and load saved remote server if available
+    this.pullSource = RemoteServer.readPreferences(tabPreferences).flatMap(view::preloadPullSource);
+    this.pullSource.ifPresent(source -> {
```

Comment:
Since the pull source should be always empty on creation, this block should be removed as well.